keras_nlp/layers/positional_embedding.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Positional embedding implementation based on `keras.layers.Layer`."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class PositionalEmbedding(keras.layers.Layer):
```

Comment:
We should come up with a naming plan for the two position embedding layers.

```
PositionalEmbedding
SinusoidalPositionalEmbedding
```

or

```
LearnedPositionalEmbedding
SinusoidalPositionalEmbedding
```

or something else?

This isn't the only was to do a positional encoding, so maybe the name is too general. On the other hand, this will be the most commonly used.

Reply:
I like 
```
LearnedPositionalEmbedding
SinusoidalPositionalEmbedding
```

Reply:
Prefer the latter one, thanks!

Reply:
Distinction to keep in mind:

- "embedding" refers to learning an "embedding space" for discrete values, i.e. mapping discrete values to vectors and learning the vector values via gradient descent
- anything else is usually referred to as "encoding", e.g. hashing encoding or multi-hot encoding.

So we have a **position embedding** layer and a **sinusoidal position encoding** layer.

Another subtlety is that "PositionalEncoding" evokes "embedding layer that takes into account positions" (a position-aware `Embedding` layer) as opposed to "a layer that embeds positions", and the name "positional embedding" typically used as such (see e.g. [this example](https://keras.io/examples/nlp/neural_machine_translation_with_transformer/#building-the-model)).

For this reason I recommend `PositionEmbedding` as the name, to navigate the distinction.

Reply:
The other one would be `CosinePositionEncoding` or `SinePositionEncoding`

Reply:
Ok, the logic makes sense. Will revert back to `PositionEmbedding`.